edx/blockstore
# Make blockstore an installable package [SE-4987]

## Description

This is a follow-up PR built on top of @symbolist's #135. It's still part of #97, and is a step towards its completion.

The previous description holds, and is copied verbatim here for reference:

- Move apps/api to apps/rest_api.
- Copy the latest code in edx-platform/openedx/core/lib/blockstore_api to blockstore/apps/api. This has been done to make the review of Blockstore as App [SE-3321] #97 easier since we will get a useful diff of changes made to each of the API methods.
- Makes the apps into an installable package.
- It also disables the migration code which changes the MySQL character set to utfmb4 to prevent accidentally running it against an edxapp database.

## Author Comments, Concerns, and Open Questions

Co-authored-by: Usman Khalid <2200617@gmail.com>

## Test Instructions

In your devstack:

- Go to your `devstack`, enable [frontend-app-library-authoring](https://github.com/edx/frontend-app-library-authoring), put everything up.
- Drop any potential blockstore tables, see:

  ```
  make dev.shell.mysql57
  mysql -u root
  DROP DATABASE blockstore_db;
  ```
  
- Provision blockstore by following its README steps.  This should create the tables from `master`.
- Run tests with `make test`.
- Go to `localhost:3001`, create a library.
- Create any block inside the library.
- Got to `blockstore` and stop blockstore development service.
- Checkout the development branch `jvdm/packagify-blockstore`.
- Provision `blockstore` one more time.
- Open `localhost:3001` and verify the library and block opens successfully.
- Add `blockstore.apps.bundles.apps.BundlesConfig`, to `lms/envs/common.py::INSTALLED_APPS`.
- Install the package in LMS:

  ```
  make dev.shell.lms
  pip install -e /edx/src/blockstore
  python manage.py lms migrate
  ```

- Open the django shell `python manage.py lms shell` and check the API methods
  can be called:

  ```
  from blockstore.api import *
  create_collection("Collection") # This should fail with requests.exceptions.ConnectionError
  ```
